src/stream/src/common/table/state_table.rs

```diff
@@ -211,6 +205,22 @@ where
             let vnode_col_idx = *idx as usize;
             pk_indices.iter().position(|&i| vnode_col_idx == i)
         });
+        let is_singleton = if vnode_col_idx_in_pk.is_some() {
```

Comment:
When streaming executors are created, the distribution is passed as `Option<Arc<Bitmap>>`, where `None` means singleton. We will use `is_none` as the ground truth for `is_singleton`, or we can even directly pass the `Option<Arc<Bitmap>>` to the `NewLocalOptions` instead of using a separate flag.

Reply:
> When streaming executors are created, the distribution is passed as `Option<Arc<Bitmap>>`, where `None` means singleton. We will use `is_none` as the ground truth for `is_singleton`, or we can even directly pass the `Option<Arc<Bitmap>>` to the `NewLocalOptions` instead of using a separate flag.

In current design, a singleton is still theoretically possible to have a non-null vnode bitmap here in the input parameters of this method. If you want to refactor, please contact with @BugenZhao